### PR TITLE
dla-future-fortran: new package

### PR DESCRIPTION
**Pull request by RMeli**
Fortran interface for [DLA-Future](https://github.com/eth-cscs/DLA-Future).

https://github.com/eth-cscs/DLA-Future-Fortran/releases/tag/v0.1.0

### CODE REVIEW

**Review by rasolca — APPROVED** [review]

**Comment by rasolca** [comment]
@spackbot fix style

**Review by msimberg — DISMISSED** [review]

File: var/spack/repos/builtin/packages/dla-future-fortran/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    generator("ninja")
+    depends_on("cmake@3.22:", type="build")
+
+    depends_on("dla-future@0.4.1: +scalapack")
+
+    depends_on("mpi", when="+test")
+    depends_on("py-fypp", when="+test", type="build")
+
+    def cmake_args(self):
```
Comment (aurianer) [inline var/spack/repos/builtin/packages/dla-future-fortran/package.py:37]:
```suggestion
    depends_on("dla-future+shared", when="+shared")
    def cmake_args(self):
        args = []
        
        args.append(self.define_from_variant("DLAF_FORTRAN_BUILD_TESTING", "test"))
```
?

File: var/spack/repos/builtin/packages/dla-future-fortran/package.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    homepage = "https://github.com/eth-cscs/DLA-Future-Fortran"
+    url = "https://github.com/eth-cscs/DLA-Future-Fortran/archive/v0.0.0.tar.gz"
+    git = "https://github.com/eth-cscs/DLA-Future-Fortran.git"
+
+    maintainers("RMeli", "rasolca", "aurianer")
+
+    license("BSD-3-Clause")
+
+    version("main")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/dla-future-fortran/package.py]: missing `branch="main"`?

Comment (RMeli) [inline var/spack/repos/builtin/packages/dla-future-fortran/package.py]:
```suggestion
    version("main", branch="main")
```

**Review by msimberg — APPROVED** [review]

**Pull request merged**